### ISSUE TITLE
Friends of a commenter can read comment even if they could not read the post

### ISSUE THREAD

**Issue by gopenshaw** [Daviddt17/island-android-app#128]

**Comment by isdavid** [comment]
What would be the benefit of this?

**Comment by gopenshaw** [comment]
Currently when a person makes a comment, all of their friends _can_ read it, though the client automatically disregards the message. If the client were modified though, someone could read all of their friend's comments, which may not be good.
For example user A says, "I'm going downtown tonight".
Then user B replies, "Awesome, John, I'll be downtown tonight too!"
Now if user C is friends with user B they could read the comment (if the code were modified) and learn about what user A is up to, even if user A is not friends with user C.

I don't think this is a major issue, but I think it might be worth addressing in a future release. If we do decide this is a problem, we could change the protocol so that each post could have a "post key" and all comments would be encrypted with the post key, in addition to the existing encryption. If we added this encryption, only the valid readers of the post could read the comment, but it would add some more complexity.

**Comment by isdavid** [comment]
Ahhh I didn't realize this was a potential issue and not a potential feature.

**Comment by gopenshaw** [comment]
Yeah that's my bad. I should have added more details when I made the issue.
